File: julia/ch1/Ex110.jl
#=
Ex110:
- Julia version: 0.6.4
- Author: Yohan
- Date: 2018-08-03
=#

module Ex110

function a(x, y)
    if y == 0
        return 0
    elseif x == 0
        return 2 * y
    elseif y == 1
        return 2
    else
        a(x - 1, a(x, y - 1))
    end

end

f(n) = a(0, n)

g(n) = a(1, n)

h(n) = a(2, n)

k(n) = 5 * n * n

end

File: octave/ch1/Ex110.m
function Ex110
  A(3,3)
  f(4)
  g(4)
  h(4)
  k(4)
endfunction

function retval = A(x, y)
  if y == 0
    retval = 0;
    return;
  elseif x == 0
    retval = 2 * y;
    return;
  elseif y == 1
    retval =  2;
    return;
  else
    retval = A(x - 1, A(x, y - 1));
  endif
endfunction

function retval = f(n)
  retval = A(0, n);
endfunction

function retval = g(n)
  retval = A(1, n);
endfunction

function retval = h(n)
  retval = A(2, n);
endfunction

function retval = k(n)
  retval = 5 * n * n;
endfunction


